### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java [hunk 1]
```diff
@@ -172,14 +175,12 @@ protected void saveResults(Resource testFile, UnitTestClassReport report) {
...
+  protected InputFile getUnitTestResource(String classKey) {
     return javaResourceLocator.findResourceByClassName(classKey);
   }
 
-  private static void saveMeasure(SensorContext context, Resource resource, Metric metric, double value) {
-    if (!Double.isNaN(value)) {
-      context.saveMeasure(resource, metric, value);
-    }
+  private static <T extends Serializable> void saveMeasure(SensorContext context, InputFile inputFile, Metric<T> metric, T value) {
+    context.<T>newMeasure().forMetric(metric).on(inputFile).withValue(value).save();
```
Comment (SonarTech) [inline java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java [hunk 2]
```diff
@@ -143,22 +144,24 @@ private void save(UnitTestIndex index, SensorContext context) {
...
-    saveMeasure(context, resource, CoreMetrics.TEST_FAILURES, report.getFailures());
-    saveMeasure(context, resource, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
+  private void save(UnitTestClassReport report, InputFile inputFile, SensorContext context) {
+    int testsCount = report.getTests() - report.getSkipped();
+    saveMeasure(context, inputFile, CoreMetrics.SKIPPED_TESTS, report.getSkipped());
+    saveMeasure(context, inputFile, CoreMetrics.TESTS, testsCount);
+    saveMeasure(context, inputFile, CoreMetrics.TEST_ERRORS, report.getErrors());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_FAILURES, report.getFailures());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
     double passedTests = testsCount - report.getErrors() - report.getFailures();
```
Comment (SonarTech) [inline java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java]: MAJOR Cast one of the operands of this substraction operation to a "double". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)

File: java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java [hunk 3]
```diff
@@ -143,22 +144,24 @@ private void save(UnitTestIndex index, SensorContext context) {
...
     if (testsCount > 0) {
-      double percentage = passedTests * 100d / testsCount;
-      saveMeasure(context, resource, CoreMetrics.TEST_SUCCESS_DENSITY, ParsingUtils.scaleValue(percentage));
+      double percentage = ParsingUtils.scaleValue(passedTests * 100d / testsCount);
+      if (!Double.isNaN(percentage)) {
+        context.<Double>newMeasure().forMetric(CoreMetrics.TEST_SUCCESS_DENSITY).on(inputFile).withValue(percentage).save();
```
Comment (SonarTech) [inline java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java]: MAJOR Make this line start at column 9. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/java/Measurer.java [hunk 4]
```diff
@@ -182,8 +180,8 @@ private static boolean isClassTree(Tree tree) {
     return tree.is(Tree.Kind.CLASS) || tree.is(Tree.Kind.INTERFACE) || tree.is(Tree.Kind.ENUM) || tree.is(Tree.Kind.ANNOTATION_TYPE);
   }
 
-  private void saveMetricOnFile(Metric metric, double value) {
-    sensorContext.saveMeasure(sonarFile, new Measure(metric, value));
+  private <T extends Serializable> void saveMetricOnFile(Metric<T> metric, T value) {
+    sensorContext.<T>newMeasure().forMetric(metric).on(sonarFile).withValue(value).save();
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/Measurer.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PublicApiChecker.java#L292](https://github.com/SonarSource/sonar-java/blob/97d3e75ab7193cc6fd9417babb6cb920d08569d7/java-frontend/src/main/java/org/sonar/java/ast/visitors/PublicApiChecker.java#L292): Cast one of the operands of this multiplication operation to a "double". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)

File: java-jacoco/src/test/java/org/sonar/plugins/jacoco/JaCoCoOverallSensorTest.java [hunk 5]
```diff
@@ -64,106 +59,106 @@ public void before() {
...
     pathResolver = mock(PathResolver.class);
-    project = mock(Project.class);
     perspectives = mock(ResourcePerspectives.class);
     sensor = new JaCoCoOverallSensor(configuration, perspectives, fileSystem, pathResolver, javaResourceLocator, javaClasspath);
   }
 
   @Test
   public void testSensorDefinition() {
-    assertThat(sensor.toString()).isEqualTo("JaCoCoOverallSensor");
+    org.fest.assertions.Assertions.assertThat(sensor.toString()).isEqualTo("JaCoCoOverallSensor");
```
Comment (m-g-sonar) [inline java-jacoco/src/test/java/org/sonar/plugins/jacoco/JaCoCoOverallSensorTest.java]: assertThat is already part of the `static import`. No need to use the fully qualified name.

File: java-jacoco/src/test/java/org/sonar/plugins/jacoco/JaCoCoOverallSensorTest.java [hunk 6]
```diff
@@ -64,106 +59,106 @@ public void before() {
...
-    verify(context).saveMeasure(eq(resource), argThat(new IsMeasure(CoreMetrics.OVERALL_UNCOVERED_LINES, 12.0)));
-    verify(context).saveMeasure(eq(resource), argThat(new IsMeasure(CoreMetrics.OVERALL_COVERAGE_LINE_HITS_DATA, "3=0;6=0;7=0;10=0;11=0;14=0;15=0;17=0;18=0;20=0;23=0;24=0")));
-    verify(context).saveMeasure(eq(resource), argThat(new IsMeasure(CoreMetrics.OVERALL_UNCOVERED_CONDITIONS, 2.0)));
+    when(configuration.shouldExecuteOnProject(false)).thenReturn(true);
+    InputFile resource = analyseReports("ut.not.found.exec", "it.not.found.exec");
+    when(configuration.shouldExecuteOnProject(false)).thenReturn(false);
+    int[] oneHitlines = new int[] {};
+    int[] zeroHitlines = new int[] {3, 6, 7, 10, 11, 14, 15, 17, 18, 20, 23, 24};
+    verifyOverallMetrics(resource, zeroHitlines, oneHitlines, 0);
+//    verify(context).saveMeasure(eq(resource), argThat(new IsMeasure(CoreMetrics.OVERALL_UNCOVERED_LINES, 12.0)));
```
Comment (m-g-sonar) [inline java-jacoco/src/test/java/org/sonar/plugins/jacoco/JaCoCoOverallSensorTest.java]: should be removed.

File: java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java [hunk 7]
```diff
@@ -86,8 +75,15 @@ public void should_register_tests() throws URISyntaxException {
     when(testCase.setType(anyString())).thenReturn(testCase);
     MutableTestPlan testPlan = mock(MutableTestPlan.class);
     when(testPlan.addTestCase(anyString())).thenReturn(testCase);
-    when(perspectives.as(eq(MutableTestPlan.class),
-        argThat(new IsResource(Scopes.FILE, Qualifiers.FILE, "ch.hortis.sonar.mvn.mc.MetricsCollectorRegistryTest")))).thenReturn(testPlan);
+    when(perspectives.as(eq(MutableTestPlan.class), argThat(new ArgumentMatcher<InputFile>() {
```
Comment (m-g-sonar) [inline java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java]: I wonder if we could not use a lambda here.

File: java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java [hunk 8]
```diff
@@ -99,132 +95,93 @@ public void should_register_tests() throws URISyntaxException {
   @Test
   public void should_store_zero_tests_when_directory_is_null_or_non_existing_or_a_file() throws Exception {
 
-    SensorContext context = mockContext();
+    SensorContextTester context = mockContext();
     parser.collect(context, null, false);
-    verify(context, never()).saveMeasure(eq(CoreMetrics.TESTS), anyDouble());
+//    verify(context, never()).saveMeasure(eq(CoreMetrics.TESTS), anyDouble());
```
Comment (m-g-sonar) [inline java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java]: If commented the test is not verifying anything anymore.

File: java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java [hunk 9]
```diff
@@ -99,132 +95,93 @@ public void should_register_tests() throws URISyntaxException {
...
+    SensorContextTester context = mockContext();
 
     parser.collect(context, getDir("multipleReports"), true);
 
     // Only 6 tests measures should be stored, no more: the TESTS-AllTests.xml must not be read as there's 1 file result per unit test
     // (SONAR-2841).
-    verify(context, times(6)).saveMeasure(argThat(new IsResource(Scopes.FILE, Qualifiers.FILE)), eq(CoreMetrics.SKIPPED_TESTS), eq(0.0));
-    verify(context, times(6)).saveMeasure(argThat(new IsResource(Scopes.FILE, Qualifiers.FILE)), eq(CoreMetrics.TESTS), anyDouble());
-    verify(context, times(6)).saveMeasure(argThat(new IsResource(Scopes.FILE, Qualifiers.FILE)), eq(CoreMetrics.TEST_ERRORS), anyDouble());
+//    verify(context, times(6)).saveMeasure(argThat(new IsResource(Scopes.FILE, Qualifiers.FILE)), eq(CoreMetrics.SKIPPED_TESTS), eq(0.0));
```
Comment (m-g-sonar) [inline java-surefire/src/test/java/org/sonar/plugins/surefire/SurefireJavaParserTest.java]: If commented the test is not verifying anything anymore.

File: java-frontend/src/test/java/org/sonar/java/MeasurerTest.java [hunk 10]
```diff
@@ -130,38 +122,29 @@ public void verify_ncloc_metric() {
...
-  private void checkMetric(boolean separateAccessorsFromMethods, File baseDir, String filename, String metric, double expectedValue) {
-    context = mock(SensorContext.class);
-    DefaultInputFile inputFile = new DefaultInputFile("", filename);
+  private void checkMetric(boolean separateAccessorsFromMethods, String filename, String metric, Number expectedValue) {
+    String relativePath = new File(baseDir, filename).getPath();
+    DefaultInputFile inputFile = new DefaultInputFile(context.module().key(), relativePath);
+    inputFile.setModuleBaseDir(fs.baseDirPath());
     fs.add(inputFile);
-    when(context.getResource(any(InputPath.class))).thenReturn(mock(Resource.class));
+//    when(context.getResource(any(InputPath.class))).thenReturn(mock(Resource.class));
```
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/MeasurerTest.java]: should be removed.

**Comment by m-g-sonar** [comment]
We should have a reference to SONARJAVA-1717 in the commit removing the cycle rule.

File: java-frontend/src/main/java/org/sonar/java/bytecode/visitor/ResourceMapping.java [hunk 11]
```diff
@@ -19,62 +19,24 @@
...
-import org.sonar.api.resources.File;
-import org.sonar.api.resources.Resource;
-
-import java.util.Collection;
-import java.util.Comparator;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Set;
 
 public class ResourceMapping {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/bytecode/visitor/ResourceMapping.java]: This whole class can probably be removed now, as it is now just a container of a Map!

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PublicApiChecker.java#L292](https://github.com/SonarSource/sonar-java/blob/83e08904bd04d88905e0bedafa8580feb3af48bb/java-frontend/src/main/java/org/sonar/java/ast/visitors/PublicApiChecker.java#L292): Cast one of the operands of this multiplication operation to a "double". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PublicApiChecker.java#L292](https://github.com/SonarSource/sonar-java/blob/d2d849872b3d0e4783411fef6b7373668529e8cb/java-frontend/src/main/java/org/sonar/java/ast/visitors/PublicApiChecker.java#L292): Cast one of the operands of this multiplication operation to a "double". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PublicApiChecker.java#L292](https://github.com/SonarSource/sonar-java/blob/6f4a84d1cf3fa3e7cab632521d1371437cc97288/java-frontend/src/main/java/org/sonar/java/ast/visitors/PublicApiChecker.java#L292): Cast one of the operands of this multiplication operation to a "double". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)

**Comment by benzonico** [comment]
Merged in https://github.com/SonarSource/sonar-java/commit/0dff1f9ce5ded1d26efe81eee00b3302633d43c3

**Pull request closed without merging**